src/VisualStudio/IntegrationTest/IntegrationTests/AbstractEditorTest.cs

```diff
@@ -78,5 +73,258 @@ protected void SetUpEditor(string markupCode)
+                }
+                else if (caretEndIndex == expectedText.Length)
+                {
+                    lineText = lineText.TrimStart();
+                }
+                else
+                {
+                    lineText = lineText.Trim();
+                }
+            }
+
+            var lineTextBeforeCaret = caretStartIndex < lineText.Length
+                ? lineText.Substring(0, caretStartIndex)
+                : lineText;
+
+            var lineTextAfterCaret = caretStartIndex < lineText.Length
+                ? lineText.Substring(caretStartIndex)
+                : string.Empty;
+
+            Assert.Equal(expectedTextBeforeCaret, lineTextBeforeCaret);
+            Assert.Equal(expectedTextAfterCaret, lineTextAfterCaret);
+            Assert.Equal(expectedTextBeforeCaret.Length + expectedTextAfterCaret.Length, lineText.Length);
+        }
+
+        protected void VerifyCurrentLineText(string expectedText, bool assertCaretPosition = false, bool trimWhitespace = true)
+        {
+            if (assertCaretPosition)
+            {
+                VerifyCurrentLineTextAndAssertCaretPosition(expectedText, trimWhitespace);
+            }
+            else
+            {
+                var lineText = Editor.GetCurrentLineText();
+
+                if (trimWhitespace)
+                {
+                    lineText = lineText.Trim();
+                }
+
+                Assert.Equal(expectedText, lineText);
+            }
+        }
+
+        private void VerifyTextContainsAndAssertCaretPosition(string expectedText)
+        {
+            var caretStartIndex = expectedText.IndexOf("$$");
+            if (caretStartIndex < 0)
+            {
+                throw new ArgumentException("Expected caret position to be specified with $$", nameof(expectedText));
+            }
+
+            var caretEndIndex = caretStartIndex + "$$".Length;
+
+            var expectedTextBeforeCaret = expectedText.Substring(0, caretStartIndex);
+            var expectedTextAfterCaret = expectedText.Substring(caretEndIndex);
+
+            var expectedTextWithoutCaret = expectedTextBeforeCaret + expectedTextAfterCaret;
+
+            var editorText = Editor.GetText();
+            Assert.Contains(expectedTextWithoutCaret, editorText);
+
+            var index = editorText.IndexOf(expectedTextWithoutCaret);
+
+            var caretPosition = Editor.GetCaretPosition();
+            Assert.Equal(caretStartIndex + index, caretPosition);
+        }
+
+        protected void VerifyTextContains(string expectedText, bool assertCaretPosition = false)
+        {
+            if (assertCaretPosition)
+            {
+                VerifyTextContainsAndAssertCaretPosition(expectedText);
+            }
+            else
+            {
+                var editorText = Editor.GetText();
+                Assert.Contains(expectedText, editorText);
+            }
+        }
+
+        protected void VerifyTextDoesNotContain(string expectedText)
```

Comment:
I think we're trying to get rid of "verify methods that just call assert in the most trivial of ways".